### PR TITLE
Use ephemeral files for `StreamSessionEvents`

### PR DESCRIPTION
**Pull request by espadolini**
The current implementation of `StreamSessionEvents` persists recordings on disk even though they're never actually reused by other instances of `StreamSessionEvents`, and periodic cleanup of session playback happens at an interval of an hour, and if the upload completer is working through a large backlog of sessions, one hour's worth of downloads might be more space that's provisioned as temporary scratch space for the Teleport instance running the Auth Service.

This PR makes it so that the downloads of `StreamSessionEvents` are placed in unlinked files in the same playback directory, which means that they'll immediately disappear as soon as the context is closed or the channels are exhausted.

changelog: improved temporary disk space usage for session recording processing

### CODE REVIEW

**Review by Tener — COMMENTED** [review]
I'm not sure if this is a right path. Wouldn't we have to re-download the sessions each time anyone requests a playback? With a big enough session this would amplify the ingress to an enormous degree.

**Comment by espadolini** [comment]
The current code does that already, except that it also interferes with the previous stream of the same session.

**Review by zmb3 — APPROVED** [review]

**Review by rosstimothy — APPROVED** [review]

File: lib/events/auditlog.go [hunk 1]
```diff
@@ -952,55 +952,47 @@ func (l *AuditLog) SearchSessionEvents(ctx context.Context, req SearchSessionEve
...
-		case <-l.ctx.Done():
-			e <- trace.BadParameter("audit log is closing, aborting the download")
-			return c, e
-		}
-	} else {
-		defer cancel()
-	}
-
-	rawSession, err := os.OpenFile(tarballPath, os.O_CREATE|os.O_RDWR|os.O_TRUNC, 0o640)
+	rawSession, err := os.CreateTemp(l.playbackDir, string(sessionID)+".stream.tar.*")
```
Comment (smallinsky) [inline lib/events/auditlog.go:959]: The following flow where tmp file is created and imedialty removed is quite hacky from reader perspective:
```
rawSession, err := os.CreateTemp(l.playbackDir, string(sessionID)+".stream.tar.*")
err := os.Remove(rawSession.Name())
// Using removed file fd
l.UploadHandler.Download(l.ctx, sessionID, rawSession)
```

Can we at least add a comment describing this behavior ?

**Review by smallinsky — APPROVED** [review]

File: lib/events/auditlog.go [hunk 2]
```diff
@@ -952,55 +952,47 @@ func (l *AuditLog) SearchSessionEvents(ctx context.Context, req SearchSessionEve
...
-		case <-downloadCtx.Done():
-		case <-l.ctx.Done():
-			e <- trace.BadParameter("audit log is closing, aborting the download")
-			return c, e
-		}
-	} else {
-		defer cancel()
-	}
-
-	rawSession, err := os.OpenFile(tarballPath, os.O_CREATE|os.O_RDWR|os.O_TRUNC, 0o640)
```
Comment (smallinsky) [inline lib/events/auditlog.go:975]: > Wait until another in progress download finishes and use its tarball.
🙃  Something went wrong here

Comment (espadolini) [reply lib/events/auditlog.go:975]: yep 😬

**Pull request merged**